### PR TITLE
Docs: fix broken link to moved content

### PR DESCRIPTION
**Pull request by imatwawana**
Fixes broken link because of moved plugins content and makes update directions for transformations more prominent.

### CODE REVIEW

**Review by jdbaldry — APPROVED** [review]

**Review by baldm0mma — APPROVED** [review]
Sorry, just saw something: in `scripts/docs/generate-transformations.ts`. Is there a reason we can't use the following syntax, like we do elsewhere?
```
[topic]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/link..."
[topic]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/link..."
```
We want to avoid having links like this `[Data frames](https://grafana.com/developers/plugin-tools/introduction/data-frames)` in the javascript content, since it's being shown verbatim in the UI. The current syntax of `[Data frames][]` will still render the link in the markdown, but not populate the UI with the text of the link itself, which is what we want. Does that make sense?

**Review by baldm0mma — CHANGES_REQUESTED** [review]
Sorry, just saw an issue. Please se my comment.

File: public/app/features/transformers/docs/content.ts [hunk 1]
```diff
@@ -966,7 +966,7 @@ export const transformationDocsContent: TransformationDocsContentType = {
     name: 'Prepare time series',
     getHelperDocs: function () {
       return `
-  Use this transformation when a data source returns time series data in a format that isn't supported by the panel you want to use. For more information about data frame formats, refer to [Data frames][].
+  Use this transformation when a data source returns time series data in a format that isn't supported by the panel you want to use. For more information about data frame formats, refer to [Data frames](https://grafana.com/developers/plugin-tools/introduction/data-frames).
```
Comment (imatwawana) [inline public/app/features/transformers/docs/content.ts:969]:
```suggestion
  Use this transformation when a data source returns time series data in a format that isn't supported by the panel you want to use. For more information about data frame formats, refer to [Data frames][].
```

**Comment by imatwawana** [comment]
> Sorry, just saw something: in `scripts/docs/generate-transformations.ts`. Is there a reason we can't use the following syntax, like we do elsewhere?
> 
> ```
> [topic]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/link..."
> [topic]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/link..."
> ```
> 
> We want to avoid having links like this `[Data frames](https://grafana.com/developers/plugin-tools/introduction/data-frames)` in the javascript content, since it's being shown verbatim in the UI. The current syntax of `[Data frames][]` will still render the link in the markdown, but not populate the UI with the text of the link itself, which is what we want. Does that make sense?

The reason I took it out is because the page had moved and this link type doesn't do the redirect even though the page is properly aliased. It basically gives us no fall-back position. I've done my best to update the link and left a note for Jack to take a look to see if I've formatted it correctly.

The directions for the page need to be updated to make clear that no inline linking can be done because that won't be immediately apparent to someone updating this page.

File: scripts/docs/generate-transformations.ts [hunk 2]
```diff
@@ -126,8 +126,8 @@ ${buildTransformationDocsContent(transformationDocsContent)}
 [Table panel]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations/visualizations/table"
 [Table panel]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations/visualizations/table"
 
-[Data frames]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/developers/plugins/introduction-to-plugin-development/data-frames"
-[Data frames]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/developers/plugins/introduction-to-plugin-development/data-frames"
+[Data frames]: "/docs/grafana/ -> /docs/developers/plugin-tools/introduction/data-frames
```
Comment (imatwawana) [inline scripts/docs/generate-transformations.ts:129]:
```suggestion
[Data frames]: "/docs/grafana/ -> /docs/developers/plugin-tools/introduction/data-frames"
```

File: scripts/docs/generate-transformations.ts [hunk 3]
```diff
@@ -126,8 +126,8 @@ ${buildTransformationDocsContent(transformationDocsContent)}
 [Table panel]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations/visualizations/table"
 [Table panel]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/panels-visualizations/visualizations/table"
 
-[Data frames]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/developers/plugins/introduction-to-plugin-development/data-frames"
-[Data frames]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/developers/plugins/introduction-to-plugin-development/data-frames"
+[Data frames]: "/docs/grafana/ -> /docs/developers/plugin-tools/introduction/data-frames)
+[Data frames]: "/docs/grafana-cloud/ -> /docs/developers/plugin-tools/introduction/data-frames"
```
[the remark this answers is not in the document]
Comment (jdbaldry) [reply scripts/docs/generate-transformations.ts]: We can use reference-style links but not `docs/reference` to link to this page.

Suggestion: 
```
[Data frames]: https://grafana.com/developers/plugin-tools/introduction/data-frames/
```

[the remark this answers is not in the document]
Comment (imatwawana) [reply scripts/docs/generate-transformations.ts]: Does that mean this link lives outside the docs/ref shortcode?

[the remark this answers is not in the document]
Comment (imatwawana) [reply scripts/docs/generate-transformations.ts]: Sorry I totally missed that this isn't part of the docs.

File: docs/README.md [hunk 4]
```diff
@@ -25,15 +25,19 @@ If you have the grafana/website repo checked out in the same directory as the gr
...
+### Transformations
+
+Auto-generated markdown location:
+
+- docs/sources/panels-visualizations/query-transform-data/transform-data/index.md
+
+Typescript location for editing and instructions:
+
+- scripts/docs/generate-transformations.ts - Includes all content not specific to a transformation.
+- public/app/features/transformers/docs/content.ts - Transformation-specific content.
```
Comment (imatwawana) [inline docs/README.md:40]:
```suggestion
- public/app/features/transformers/docs/content.ts - Transformation-specific content.

Only use [reference style links](https://grafana.com/docs/writers-toolkit/write/shortcodes/#docsreference) in the `content.ts` file or else link text will be visible in the UI.
```

File: docs/sources/panels-visualizations/query-transform-data/transform-data/index.md [hunk 5]
```diff
@@ -1,6 +1,12 @@
 ---
 comments: |
-  This Markdown file is auto-generated. Do not edit this file directly.
+  This Markdown file is auto-generated. DO NOT EDIT THIS FILE DIRECTLY.
+  
+  To update this Markdown, navigate to the following Typescript files and edit them based on what you need to update:
+
+  scripts/docs/generate-transformations.ts - Includes all content not specific to a transformation.
+  public/app/features/transformers/docs/content.ts - Transformation-specific content.
+
```
Comment (imatwawana) [inline docs/sources/panels-visualizations/query-transform-data/transform-data/index.md:9]:
```suggestion
  
  Only use reference style links in the `content.ts` file or else link text will be visible in the UI.
```

**Comment by baldm0mma** [comment]
> > Sorry, just saw something: in `scripts/docs/generate-transformations.ts`. Is there a reason we can't use the following syntax, like we do elsewhere?
> > ```
> > [topic]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/link..."
> > [topic]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/link..."
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We want to avoid having links like this `[Data frames](https://grafana.com/developers/plugin-tools/introduction/data-frames)` in the javascript content, since it's being shown verbatim in the UI. The current syntax of `[Data frames][]` will still render the link in the markdown, but not populate the UI with the text of the link itself, which is what we want. Does that make sense?
> 
> The reason I took it out is because the page had moved and this link type doesn't do the redirect even though the page is properly aliased. It basically gives us no fall-back position. I've done my best to update the link and left a note for Jack to take a look to see if I've formatted it correctly.
> 
> The directions for the page need to be updated to make clear that no inline linking can be done because that won't be immediately apparent to someone updating this page.

Awesome! Thanks! As long as @jdbaldry is happy with the new link, I am also. I just wanted to make sure the link text stayed out of the UI render.

**Review by baldm0mma — DISMISSED** [review]

File: scripts/docs/generate-transformations.ts [hunk 6]
```diff
@@ -151,6 +156,8 @@ ${buildTransformationDocsContent(transformationDocsContent)}
 [dashboard variable]: "/docs/grafana-cloud/ -> docs/grafana/<GRAFANA VERSION>/dashboards/variables"
 
 {{% /docs/reference %}}
+
+[Data frames]: https://grafana.com/developers/plugin-tools/introduction/data-frames/
```
Comment (baldm0mma) [inline scripts/docs/generate-transformations.ts:160]: @imatwawana I added your changes to the `scripts/docs/generate-transformations.ts` file, and then built the markdown (docs/sources/panels-visualizations/query-transform-data/transform-data/index.md) from there, but it looks like the link in the markdown for `Data frames` doesn't show up after building. Should be on line 922 in the index.md, but isn't. I don't know if this syntax will work.

**Comment by baldm0mma** [comment]
> > > Sorry, just saw something: in `scripts/docs/generate-transformations.ts`. Is there a reason we can't use the following syntax, like we do elsewhere?
> > > ```
> > > [topic]: "/docs/grafana/ -> /docs/grafana/<GRAFANA VERSION>/link..."
> > > [topic]: "/docs/grafana-cloud/ -> /docs/grafana/<GRAFANA VERSION>/link..."
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > We want to avoid having links like this `[Data frames](https://grafana.com/developers/plugin-tools/introduction/data-frames)` in the javascript content, since it's being shown verbatim in the UI. The current syntax of `[Data frames][]` will still render the link in the markdown, but not populate the UI with the text of the link itself, which is what we want. Does that make sense?
> > 
> > 
> > The reason I took it out is because the page had moved and this link type doesn't do the redirect even though the page is properly aliased. It basically gives us no fall-back position. I've done my best to update the link and left a note for Jack to take a look to see if I've formatted it correctly.
> > The directions for the page need to be updated to make clear that no inline linking can be done because that won't be immediately apparent to someone updating this page.
> 
> Awesome! Thanks! As long as @jdbaldry is happy with the new link, I am also. I just wanted to make sure the link text stayed out of the UI render.

Looks like that new link syntax isn't working. :(

File: scripts/docs/generate-transformations.ts (hunk 6 above)
Comment (baldm0mma) [reply scripts/docs/generate-transformations.ts:160]: But it does seem to render in the html, so maybe we're good? 

Screenshot 2023-10-25 at 11 15 54 AM

**Comment by jdbaldry** [comment]
It's working for me though I have to fix the Makefile in order to run the doc generation.

**Comment by baldm0mma** [comment]
> It's working for me though I have to fix the Makefile in order to run the doc generation.

Why was this again? Works for me. Was it an issue with our different CLIs?

**Comment by baldm0mma** [comment]
> > It's working for me though I have to fix the Makefile in order to run the doc generation.
> 
> Why was this again? Works for me. Was it an issue with our different CLIs? And is this something we need to remedy?

@jdbaldry i updated to GNU Make 4.4.1, and you're update to the make file works great!

**Review by baldm0mma — APPROVED** [review]
Great!

**Pull request merged**